blockchain/types/derive_sha.go

```diff
@@ -40,9 +41,20 @@ const (
 )
 
 var (
+	commonEmptyRootHash = common.HexToHash("0x56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421")
+
 	// EmptyRootHashOriginal is the empty root hash of a state trie,
 	// which is equal to EmptyRootHash with ImplDeriveShaOriginal.
-	EmptyRootHashOriginal = common.HexToHash("0x56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421")
+	EmptyRootHashOriginal = commonEmptyRootHash
+
+	// EmptyTxsHash is the known hash of the empty transaction set.
+	EmptyTxsHash = commonEmptyRootHash
+
+	// EmptyReceiptsHash is the known hash of the empty receipt set.
+	EmptyReceiptsHash = commonEmptyRootHash
+
+	// EmptyCodeHash is the known hash of the empty EVM bytecode.
+	EmptyCodeHash = crypto.Keccak256Hash(nil) // c5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470
```

Comment:
Can we have these names instead?

```
// Root hash of an empty trie.
// Used to represent a block's empty StateRoot or an account's StorageRoot.
EmptyRootHash = 0x56e8...

// Hash of an empty byte array.
// Used to represent an account's empty contract code.
EmptyCodeHash = 0xc5d2..

// Empty TransactionRoot and ReceiptsRoot for the Original algorithm.
EmptyTxRootOriginal = EmptyRootHash

// Empty TransactionRoot and ReceiptsRoot for the Simple algorithm.
EmptyTxRootSimple = EmptyCodeHash

// Empty TransactionRoot and ReceiptsRoot for the Simple algorithm.
EmptyTxRootConcat = EmptyCodeHash
```

Reply:
Nice suggestion. The name `EmptyRootHash` has already been declared in L63.
We have two options; changing the function name of L63 or changing the suggested `EmptyRootHash` to another. Which one do you prefer?